posts/GSoC 2019.md

```diff
@@ -0,0 +1,64 @@
+---
+title: GSoC 2019 PyCanChemAndYouCan2
+date: 2019-06-07 12:33:00 UTC-16:33
+slug: GSoC 2019
+tags: GSoC 2019
+description:  2019 Google Summer of Code Cnatera Project Blog 
+type: text
+author: Chao Xu
+
+---
+
+# GSoC 2019 PyCanChemAndYouCan2 (ChemCheck)
+Hi everyone! I'm Chao Xu, a master student in Chemical Engineering at Northeastern University. I am working on a **Cantera** project in GSoC 2019 with my mentor Richard West([@rwest](https://github.com/rwest)) and other Cantera committee members([@bryanwweber](https://github.com/bryanwweber), [@KyleLinevitchJr](https://github.com/KyleLinevitchJr)). I am glad to introduce my project here!
+
+## What is PyCanChemAndYouCan2(ChemCheck)
+
+PyCanChemAndYouCan2 is a web-based debugging tool developed for Cantera to visualize and diagnose syntax errors and chemical errors in input files. Users can upload  input files and visualize their errors on the website, and they can also download it after editing the input files.
+
+## Why Would I Need It
+
+Cantera uses input files containing mechanism of reactions because mechanism information is the fundamental to do thermo-kinetic calculation. The input files are required to be in .cti format, and users are able to gain them from various sources (e.g.: writing .cti file by hands, adopted from various sources and repositories on the web, or via a hybrid of these two approaches etc.).   However, .cti files always include errors since they are from different sources, which could cause  inaccurate results. The errors include syntax errors and chemical errors, and it is not likely to be found manually especially in large mechanism files. In this case, a debugging tool helping users to find and fix errors is very useful. 
+## What Errors Can Be Visualized or Fixed:
+The errors being detected by the application will include but not limited:
+Syntax errors
+
+-   duplicate lines
+-   missing digits from end of line
+-   missing 'E'
+-   whatever else we find
+
+Chemistry errors
+
+-   discontinuous polynomicals
+-   exceeding collision limit
+-   inconsistent thermo for explicit reverse rates
+-   negative rate coefficient (as a  [result](https://github.com/comocheng/wiki/issues/375#)  of sum of arrhenius or PLOG expressions, eg.  [here](https://github.com/Cantera/cantera-website/issues/77))
+-   excess stiffness
```

Comment:
This one and the next two aren't really chemistry errors, so much as general problems. I would classify them into a new category.

Reply:
You also need to run a simulation to catch them, and even then, the simulation would have to be run under some particular conditions. I'm not sure this is something the website can do very easily. However, making it easier to understand the errors in Cantera is very valuable.